templates/terraform/examples/region_network_endpoint_group_basic.tf.erb

```diff
@@ -0,0 +1,121 @@
+// Cloud Run Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "run-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  cloud_run {
+    service = google_cloud_run_service.<%= ctx[:primary_resource_id] %>.name
+  }
+}
+
+resource "google_cloud_run_service" "<%= ctx[:primary_resource_id] %>" {
+  name     = "hello-world"
+  location = "us-central1"
+
+  template {
+    spec {
+      containers {
+        image = "gcr.io/cloudrun/hello"
+      }
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+}
+
+
+// Cloud Functions Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "gcf-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  cloud_function {
+    function = google_cloudfunctions_function.<%= ctx[:primary_resource_id] %>.name
+  }
+}
+
+resource "google_cloudfunctions_function" "<%= ctx[:primary_resource_id] %>" {
+  name        = "function-test"
+  description = "My function"
+  runtime     = "nodejs10"
+
+  available_memory_mb   = 128
+  source_archive_bucket = google_storage_bucket.bucket.name
+  source_archive_object = google_storage_bucket_object.archive.name
+  trigger_http          = true
+  entry_point           = "helloGET"
+}
+
+// App Engine Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "gae-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  app_engine {
+    service = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.service
+    version = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.version_id
+  }
+}
+
+resource "google_app_engine_flexible_app_version" "<%= ctx[:primary_resource_id] %>" {
+  version_id = "v1"
+  service    = "default"
+  runtime    = "nodejs"
+
+  entrypoint {
+    shell = "node ./app.js"
+  }
+
+  deployment {
+    zip {
+      source_url = "https://storage.googleapis.com/${google_storage_bucket.bucket.name}/${google_storage_bucket_object.object.name}"

```

Comment:
Does `google_storage_bucket_object.object.self_link` work here?

Reply:
I don't know, copied from the gae flex example

Reply:
pushed a commit the selflink, lets see what it does

Reply:
Please try to run the tests yourself- we don't run them automatically on PRs due to security concerns. I'm happy to run them for you, but it'll take a lot longer to go back and forth that way than it would if you did it locally :)